active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,234 @@
+runtimeclass NumberBoxValueChangedEventArgs
+{
+    String Text;
+    Double Value
+};
+
+runtimeclass NumberBoxTextChangedEventArgs
+{
+    String Text;
+    Double Value
+};
+
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    Double Value;
+    Boolean AcceptsCalculation;
+    
+    Boolean UpDownButtonsEnabled;
+    Boolean HyperDragEnabled;
+    Boolean HyperScrollEnabled;
+    Double StepFrequency;
+  
+    Boolean AreLeadingZerosTrimmed;
+    Double DecimalPrecision;
+    Boolean DoesInputRound;
+    
+    Double MinValue;
+    Double MaxValue;
+    
+    Boolean IsInvalidInputOverwritten
+    
+
+    NumberBoxTemplateSettings TemplateSettings{ get; };
+
+    event Windows.Foundation.TypedEventHandler<NumberBox, NumberBoxValueChangedEventArgs> ValueChanged;
+    event Windows.Foundation.TypedEventHandler<NumberBox, NumberBoxTextChangedEventArgs> TextChanged;
+
+    static Windows.UI.Xaml.DependencyProperty ValueProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty AcceptsCalculationProperty{ get; };
+    
+    static Windows.UI.Xaml.DependencyProperty UpDownButtonsEnabledProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty HyperDragEnabledProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty HyperScrollEnabledProperty{ get; };   
+    static Windows.UI.Xaml.DependencyProperty StepFrequencyProperty{ get; };
+    
+    static Windows.UI.Xaml.DependencyProperty AreLeadingZerosTrimmedProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty DecimalPrecisionProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty DoesInputRoundProperty{ get; };
+    
+    static Windows.UI.Xaml.DependencyProperty MinValueProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MaxValueProperty{ get; };
+    
+    static Windows.UI.Xaml.DependencyProperty IsInvalidInputOverwrittenProperty{ get; };
+}
+```
+
+## Appendix
+<!-- Anything else that you want to write down for posterity, but 
+that isn't necessary to understand the purpose and usage of the API.
+For example, implementation details. -->
+
+### Behavioral Components
+
+| Property | Notes |
+|:---:|:---|
+| Value/Text Changed | * When Text is changed via codebehind or by user input ("Enter" is pressed or the NumberBox loses focus) the NumberBox will be prompted to update its Text and Value properties. A TextChanged event will be raised that will exposes the new Text and the old Value and allows the developer to intercept the and manipulate these properties (such as for manually handling validation error). Value will then be updated to the Text input converted to a Double. <br><br> * Conversely, code behind updates to Value will raise a ValueChanged event that will exposes the new Value and the old Text and allows the developer to intercept the and manipulate these properties (such as for manually handling validation error). Text will then be updated to the Value input converted to a String.    |
```

Comment:
Scenarios corresponding to error states would be great to explicitly call out and have a point of view on. I'm still not sure if/how we handle error states - is it an exception? A piece of UI? How does error states affect the "Value" property? How do they work with the input validation stuff mentioned here? 

Specific error states that are on my mind are:
* Div by zero
* Min or Max reached/exceeded
* Incorrect text entered
* Unsupported operator

Reply:
For Min or Max values, rather than returning the value to the previous one, there may be a case to clamp values above to the Max value, or values below to the Min value.

Reply:
Probably. I can see a bunch of potential possibilities including: 
* clamp to the min/max
* return to last known good value
* wrap around

I think we need to figure out a default behavior and then let the dev have the capability to do the others.